tests/foreman/api/test_hostcollection.py

```diff
@@ -23,438 +23,464 @@
-        """Create host collections with different descriptions.
-
-        :id: 9d13392f-8d9d-4ff1-8909-4233e4691055
-
-        :expectedresults: The host collection was successfully created and has
-            appropriate description.
-
-        :CaseImportance: Critical
-        """
-        for desc in valid_data_list():
-            with self.subTest(desc):
-                host_collection = entities.HostCollection(
-                    description=desc, organization=self.org
-                ).create()
-                self.assertEqual(host_collection.description, desc)
-
-    @pytest.mark.tier1
-    def test_positive_create_with_limit(self):
-        """Create host collections with different limits.
-
-        :id: 86d9387b-7036-4794-96fd-5a3472dd9160
-
-        :expectedresults: The host collection was successfully created and has
-            appropriate limit.
-
-        :CaseImportance: Critical
-        """
-        for limit in (1, 3, 5, 10, 20):
-            with self.subTest(limit):
-                host_collection = entities.HostCollection(
-                    max_hosts=limit, organization=self.org
-                ).create()
-                self.assertEqual(host_collection.max_hosts, limit)
-
-    @pytest.mark.tier1
-    def test_positive_create_with_unlimited_hosts(self):
-        """Create host collection with different values of 'unlimited hosts'
-        parameter.
-
-        :id: d385574e-5794-4442-b6cd-e5ded001d877
-
-        :expectedresults: The host collection was successfully created and has
-            appropriate 'unlimited hosts' parameter value.
-
-        :CaseImportance: Critical
-        """
-        for unlimited in (True, False):
-            with self.subTest(unlimited):
-                host_collection = entities.HostCollection(
-                    max_hosts=1 if not unlimited else None,
-                    organization=self.org,
-                    unlimited_hosts=unlimited,
-                ).create()
-                self.assertEqual(host_collection.unlimited_hosts, unlimited)
-
-    @pytest.mark.tier1
-    def test_positive_create_with_host(self):
-        """Create a host collection that contains a host.
-
-        :id: 9dc0ad72-58c2-4079-b1ca-2c4373472f0f
-
-        :expectedresults: The host collection can be read back, and it includes
-            one host.
-
-        :CaseImportance: Critical
+    :CaseImportance: Critical
+    """
+    host_collection = entities.HostCollection(description=desc, organization=module_org).create()
+    assert host_collection.description == desc
 
-        :BZ: 1325989
-        """
+
+@pytest.mark.tier1
+def test_positive_create_with_limit(module_org):
+    """Create host collections with different limits.
+
+    :id: 86d9387b-7036-4794-96fd-5a3472dd9160
+
+    :expectedresults: The host collection was successfully created and has
+        appropriate limit.
+
+    :CaseImportance: Critical
+    """
+    for limit in (1, 3, 5, 10, 20):
```

Comment:
```suggestion
    for _ in range(5):
          limit=random.randint(1,30)   
```

Reply:
OK, thanks

Reply:
@devendra104 Changes made, please review once more. Thank you